birdhouse/optional-components/README.md

```diff
@@ -129,3 +148,44 @@ How to enable in `env.local` (a copy from
 * Add `./optional-components/all-public-access` to `EXTRA_CONF_DIRS`.
 
 The anonymous user will now have all the permissions described in [`./optional-components/all-public-access/all-public-access-magpie-permission.cfg`](all-public-access/all-public-access-magpie-permission.cfg).
+
+
+## Weaver 
```

Comment:
I think Weaver will become pretty much an essential component for PAVICS stack?  In this case, better put in in `birdhouse/components`.

There are `components/` and `optional-components/` because as previously discussed when I first introduced this pluggable components mechanism, I was planning to break up the root `docker-compose.yml` into many components as well, leaving only the proxy in the root `docker-compose.yml`.  So all those previously in the root `docker-compose.yml` will go into `components/`.

`optional-components/` are for nice to have, test helper, optional components, like `testthredds`, `emu`, `all-public-access` and so on.

Reply:
We tried to be nice here and don't push weaver too far until it is fully integrated in the stack (with birdy for example). If you don't mind to have that component meanwhile, we can put it in components.

Reply:
But just to be clear the mecanism for components and optional-components is the same in that they are both pluggable?

Reply:
> But just to be clear the mecanism for components and optional-components is the same in that they are both pluggable?

Exact, both are pluggable and not enabled by default.  Stuff in `components/` just "look" more essential and have better README documentation (auto gen table of content).

Reply:
Moved.